File: 2685. Count the Number of Complete Components.cpp
class Solution {
public:
    void dfs(int node,
             vector<vector<int>>& adj,
             vector<bool>& vis,
             vector<int>& component) {

        vis[node] = true;
        component.push_back(node);

        for (int nei : adj[node]) {
            if (!vis[nei]) {
                dfs(nei, adj, vis, component);
            }
        }
    }
    
    int countCompleteComponents(int n, vector<vector<int>>& edges) {
        vector<vector<int>> adj(n);
        vector<int> degree(n,0);

        for(auto &edge:edges)
        {
            int u=edge[0];
            int v=edge[1];

            adj[u].push_back(v);
            adj[v].push_back(u);

            degree[u]++;
            degree[v]++;
        }

        vector<bool> vis(n,false);
        int ans=0;

        for(int i=0;i<n;i++)
        {
            if(vis[i])
                continue;
            
            vector<int> component;
            dfs(i,adj,vis,component);

            int sz = component.size();

            bool complete = true;

            for (int node : component) {
                if (degree[node] != sz - 1) {
                    complete = false;
                    break;
                }
            }

            if (complete)
                ans++;
        }
        return ans;
    }
};

File: 2685. Count the Number of Complete Components.py
from typing import List

class Solution:
    def dfs(self, node, adj, vis, component):
        vis[node] = True
        component.append(node)

        for nei in adj[node]:
            if not vis[nei]:
                self.dfs(nei, adj, vis, component)

    def countCompleteComponents(self, n: int, edges: List[List[int]]) -> int:
        adj = [[] for _ in range(n)]
        degree = [0] * n

        # Build graph and degree array
        for u, v in edges:
            adj[u].append(v)
            adj[v].append(u)

            degree[u] += 1
            degree[v] += 1

        vis = [False] * n
        ans = 0

        for i in range(n):
            if vis[i]:
                continue

            component = []
            self.dfs(i, adj, vis, component)

            size = len(component)
            complete = True

            for node in component:
                if degree[node] != size - 1:
                    complete = False
                    break

            if complete:
                ans += 1

        return ans